airflow/models.py

```diff
@@ -920,6 +920,17 @@ def is_queueable(
         else:
             return False
 
+
+    def is_premature(self):
+        """
+        Returns whether a task is in UP_FOR_RETRY state and its retry interval
+        has elapsed.
+        """
+        # is the task still in the retry waiting period?
+        if self.state == State.UP_FOR_RETRY and not self.ready_for_retry():
```

Comment:
Can't this whole method just be:

    return self.state == State.UP_FOR_RETRY and not self.ready_for_retry()